README.md

```diff
@@ -1,408 +1,41 @@
-linux-system-roles Logging (LSR/Logging)
-========================================
+# linux-system-roles/logging
 
-Guidelines for Using Logging Ansible Role
------------------------------------------
+## Background
 
-The `logging` role enables a RHEL admin/developer to deploy logging collectors on the local host, remote host or set of remote hosts,
-process these logs if needed to add additional metadata and ship it to a remote location to be saved and analyzed.
+Logging Role is an abstract layer for provisioning and configuring the logging system. Currently, rsyslog is only a supported provider.
 
-The logging role currently supports `Rsyslog` as the log collector.
+In the nature of logging, there are multiple ways to read logs and multiple ways to output them. For instance, the logging system may read logs from local files, or read them from systemd/journal, or receive them from the other logging system over the network. Then, the logs may be stored in the local files in the /var/log directory, or sent to Elasticsearch, or forwarded to other logging system. The combination between the inputs and the outputs needs to be flexible. For instance, you may want to inputs from journal stored just in the local file, while inputs read from files stored in the local log files as well as forwarded to the other logging system.
 
-Table of Contents
-=================
+To satisfy such requirements, logging role introduced 3 primary parameters `logging_inputs`, `logging_outputs`, and `logging_flows`. The inputs are represented in the list of `logging_inputs` dictionary, the outputs are in the list of `logging_outputs` dictionary, and the relationship between them are defined as a list of `logging_flows` dictionary. The details are described in [Rsyslog](roles/rsyslog).
 
-<!--ts-->
-   * [linux-system-roles Logging](#linux-system-roles-logging)
-   * [Table of contents](#table-of-contents)
-   * [Definitions](#definitions)
-   * [Deploy Default Logging Configuration Files](#deploy-default-logging-configuration-files)
-   * [Deploy Configuration Files](#deploy-configuration-files)
-      * [Inventory File](#inventory-file)
-      * [vars.yml](#varsyml)
-      * [Variables in vars.yml](#variables-in-varsyml)
-      * [playbook.yml](#playbookyml)
-   * [New Projects Integration](#new-projects-integration)
-      * [Planned Flows](#planned-flows)
-   * [Testing](#testing)
-   * [Additional Resources](#additional-resources)
-   * [License](#license)
-<!--te-->
+## Definitions
 
-Definitions
```

Comment:
You need to have a section that specifies the variables used as input parameters to the logging role - see https://github.com/linux-system-roles/firewall#variables - you can also have a `Definitions` section where you define the terms used.  You should also have a section https://github.com/linux-system-roles/firewall#example-playbooks with some short examples.

Reply:
> You need to have a section that specifies the variables used as input parameters to the logging role - see https://github.com/linux-system-roles/firewall#variables - you can also have a `Definitions` section where you define the terms used. You should also have a section https://github.com/linux-system-roles/firewall#example-playbooks with some short examples.

Somehow I put some of them (maybe not all though) in the rsyslog README. Do you think I should move them to the top README?

Reply:
> Do you think I should move them to the top README?

Yes.  All public parameters should be in the top level `logging/README.md`, and should be specified in the top level `logging/defaults/main.yml`.